### PR TITLE
Filter entities with no area configured

### PR DESCRIPTION
**Pull request by nivnoach**
Add an option in the Entities page to show only entities with no area configured.
This fulfills feature request @ https://github.com/home-assistant/frontend/discussions/7916

## Proposed change
In Entities page, add another option to the filter drop-down, to show 'entities configured with area'. 
By default this will be on to preserve backwards-compatibility.
When unchecked, only entities with *no* area configured will be visible - allowing quick and easy setup of the system. 

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion: https://github.com/home-assistant/frontend/discussions/7916
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/entities/ha-config-entities.ts [hunk 1]
```diff
@@ -363,6 +366,10 @@ export class HaConfigEntities extends SubscribeMixin(LitElement) {
           continue;
         }
 
+        if (!showAreaConfigured && areaId) {
```
Comment (nivnoach) [inline src/panels/config/entities/ha-config-entities.ts:369]: I am still not sure if this should be `area` or `areaId`.

File: src/translations/en.json [hunk 2]
```diff
@@ -1854,6 +1854,7 @@
               "filter": "Filter",
               "show_disabled": "Show disabled entities",
               "show_unavailable": "Show unavailable entities",
+              "show_with_area": "Show etities with area configured",
```
Comment (spacegaier) [inline src/translations/en.json:1857]:
```suggestion
              "show_with_area": "Show entities with area configured",
```
Or perhaps even better "Show entities with assigned area". Reason: The area is not itself configured here, but just assigned / set, so "configured" sounds weird to me here.

File: translations/frontend/en.json [hunk 3]
```diff
@@ -1782,7 +1782,8 @@
                             "show_all": "Show all",
                             "show_disabled": "Show disabled entities",
                             "show_readonly": "Show read-only entities",
-                            "show_unavailable": "Show unavailable entities"
+                            "show_unavailable": "Show unavailable entities",
+							"show_with_area": "Show etities with area configured"
```
Comment (spacegaier) [inline translations/frontend/en.json:1786]: Do not make changes to the translation files. Those are handled via "lokalise" project.

**Comment by spacegaier** [comment]
BTW: To completely fulfill the mentioned FR this also needs to be added to the "Device" page. But probably best to do that in a separate PR.

File: src/translations/en.json (hunk 2 above)
Comment (nivnoach) [reply src/translations/en.json:1857]: Agree!

File: translations/frontend/en.json (hunk 3 above)
Comment (nivnoach) [reply translations/frontend/en.json:1786]: Without this, I get an empty item in the menu
image

Comment (spacegaier) [reply translations/frontend/en.json]: The `script/develop` should be taking care of adding them to the EN translation file in GULP step `build-master-translation`.

File: src/panels/config/entities/ha-config-entities.ts (hunk 1 above)
Comment (bramkragten) [reply src/panels/config/entities/ha-config-entities.ts:369]: `areaId` is fine here

File: translations/frontend/en.json [hunk 4]
```diff
@@ -1782,7 +1782,8 @@
                             "show_all": "Show all",
                             "show_disabled": "Show disabled entities",
                             "show_readonly": "Show read-only entities",
-                            "show_unavailable": "Show unavailable entities"
+                            "show_unavailable": "Show unavailable entities",
+							"show_with_area": "Show entities with assigned area"
```
Comment (bramkragten) [inline translations/frontend/en.json:1786]: We use spaces, not tabs
```suggestion
                    "show_with_area": "Show entities with assigned area"
```

**Comment by bramkragten** [comment]
I'm not too sure this is needed, you could sort the area column to get the entities without an area bundled together?

**Pull request closed without merging**